### ISSUE TITLE
Support for serialport v6

### ISSUE THREAD

**Issue by ekawahyu** [reyiyo/virtual-serialport#20]
When would be the timeline for virtual-serialport to support serialport v6? According to node-serialport documentation, 'data' event is not a preferable way to read from the serial port. Instead, it suggests to use .read().

So, when I use other libraries that support serialport v6, they mostly don't listen to 'data' event anymore. The virtual-serialport is broken in this regard. I have no idea how to move forward to fix this issue.

**Comment by ekawahyu** [comment]
Here is the recently updated h5.modbus with support only for serialport > 6:
https://github.com/ekawahyu/h5.modbus/tree/master

I made a branch to support 'data' event temporarily here:
https://github.com/ekawahyu/h5.modbus/tree/fix/virtual-serialport-support

Please help to make some necessary changes to make it work with the latest h5.modbus. Thank you.

**Comment by Apollon77** [comment]
Also the parsers have changed and a lot more ...

**Comment by Apollon77** [comment]
Additionally: Maybe an update for serialport 6.x is not needed at all ...
Serialport 6 has a "Mocking" feature where you can define a port and can interact in a comparable way with serialport directly.
Currently you need to have direct access to the initialized Serialport instance, but maybe they change it (i opened an issue with them or this).

**Comment by Lucas-C** [comment]
> Serialport 6 has a "Mocking" feature where you can define a port and can interact in a comparable way with serialport directly.

Indeed, this lib is no longer necesseray: https://serialport.io/docs/api-binding-mock

**Comment by Apollon77** [comment]
@Lucas-C Really?

What about
> Currently you need to have direct access to the initialized Serialport instance, but maybe they change it (i opened an issue with them or this).

I think this is even not changed in the MockBinding

**Comment by Lucas-C** [comment]
I introduced a `MockBinding.getOpenMockPort()` method in this PR: https://github.com/serialport/binding-mock/pull/20
I use it on my project with [patch-package](https://www.npmjs.com/package/patch-package) to get access to the `MockPortBinding`.
Just sharing, maybe that won't answer your need, but I hope it can help 😊

**Comment by Apollon77** [comment]
That indeed could work
